inf-clojure.el

```diff
@@ -843,6 +847,115 @@ HOST is the host the process is running on, PORT is where it's listening."
+  (when (not (process-live-p process))
+    (let ((repl-buffer (process-buffer process)))
+      (with-current-buffer repl-buffer
+        (when inf-clojure-socket-buffer
+          (kill-buffer inf-clojure-socket-buffer))))))
+
+(defvar inf-clojure-socket-repl-startup-forms
+  '((lein . "JVM_OPTS='-Dclojure.server.repl={:port %s :accept clojure.core.server/repl}' lein repl")
+    (boot . "export BOOT_JVM_OPTIONS='-Dclojure.server.repl=\"{:port %s :accept clojure.core.server/repl}\"' boot repl")
+    (clojure . "clojure -J-Dclojure.server.repl=\"{:port %s :accept clojure.core.server/repl}\"")
+    (cljs . "clojure -J-Dclojure.server.repl=\"{:port %s :accept cljs.server.browser/repl}\"")
+    (lein-clr . "JVM_OPTS='-Dclojure.server.repl={:port %s :accept clojure.core.server/repl}' lein clr repl")
+    (planck . "planck -n %s")
+    (babashka . "bb socket-repl %s")))
+
+(defcustom inf-clojure-socket-repl-port
+  nil
+  "Port to be used when creating a socket REPL via `inf-clojure-socket-repl'.
+If left as nil a random port will be selected between 5500-6000."
+  :type '(choice integer (const nil)))
+
+;;;###autoload
+(defun inf-clojure-socket-repl (cmd)
+  "Start a socket REPL server and connect to it via `inf-clojure'.
+CMD is the command line used to start the socket REPL, if this
+isn't provided you will be prompted to select from the defaults
+provided in `inf-clojure-socket-repl-startup-forms' or
+`inf-clojure-custom-startup' if this is defined."
+  (interactive (list (or (unless current-prefix-arg
+                           inf-clojure-custom-startup)
+                         (completing-read "Select Clojure REPL startup command: "
+                                          (mapcar #'cdr inf-clojure-socket-repl-startup-forms)
+                                          nil
+                                          'confirm-after-completion))))
+  (let* ((host "localhost")
+         (port (or inf-clojure-socket-repl-port (+ 5500 (random 500))))
+         (repl-type (or (unless prefix-arg
+                          inf-clojure-custom-repl-type)
+                        (car (rassoc cmd inf-clojure-socket-repl-startup-forms))
+                        (inf-clojure--prompt-repl-type)))
+         (project-name (inf-clojure--project-name (or (clojure-project-dir) "standalone")))
+         (socket-process-name (format "*%s-%s-socket-server*" project-name repl-type))
+         (socket-buffer-name (format "*%s-%s-socket*" project-name repl-type))
+         (socket-buffer (get-buffer-create socket-buffer-name))
+         (repl-buffer-name (format "%s-%s-repl" project-name repl-type))
+         (socket-form (or (cdr (assoc repl-type inf-clojure-socket-repl-startup-forms))
+                          inf-clojure-custom-startup
+                          cmd))
+         (socket-cmd (format socket-form (number-to-string port)))
+         (sock (let ((default-directory (or (clojure-project-dir) default-directory)))
```

Comment:
I've noticed this is the second `clojure-project-dir` call in the big `let`, so you might want to factor it out as a `let` binding.